docs/how-to/hip_runtime_api/memory_management/host_memory.rst

```diff
@@ -0,0 +1,205 @@
+  hipMalloc((int **)&device_output, ELEMENT_NUMBER * sizeof(int));
+
+  // Device data preparation
+  hipMemcpy(device_input, host_input, ELEMENT_NUMBER * sizeof(int), cudaMemcpyHostToDevice);
+  hipMemset()
+
+  // Run the kernel
+  ...
+
+  hipMemcpy(device_input, host_input, ELEMENT_NUMBER * sizeof(int), cudaMemcpyDeviceToHost);
+
+  // Free host memory
+  delete[] host_input;
+  delete[] host_output;
+
+  // Free device memory
+  hipFree()
+  hipFree()
+
+.. note::
+
+  :cpp:func:`hipMalloc` and :cpp:func:`hipFree` are blocking calls, however, HIP
+  recently added non-blocking versions :cpp:func:`hipMallocAsync` and
+  :cpp:func:`hipFreeAsync` which take in a stream as an additional argument.
+
+Pinned memory
+================================================================================
+
+Pinned memory (or page-locked memory, or non-pageable memory) is host memory
+that is mapped into the address space of all GPUs, meaning that the pointer can
+be used on both host and device. Accessing host-resident pinned memory in device
+kernels is generally not recommended for performance, as it can force the data
+to traverse the host-device interconnect (e.g. PCIe), which is much slower than
+the on-device bandwidth (>40x on MI200).
+
+Much like how a process can be locked to a CPU core by setting affinity, a
+pinned memory allocator does this with the memory storage system. On multi-socket
+systems it is important to ensure that pinned memory is located on the same
+socket as the owning process, or else each cache line will be moved through the
+CPU-CPU interconnect, thereby increasing latency and potentially decreasing
+bandwidth.
+
+Advantage of pinned memory is the improved transfer times between host and
+device. For transfer operations, such as :cpp:func:`hipMemcpy` or :cpp:func:`hipMemcpyAsync`,
+using pinned memory instead of pageable memory on host can lead to a ~3x
+improvement in bandwidth.
+
+Disadvantage of pinned memory that it reduces the available RAM for other
+operations, such as paging, which can negatively impact the overall performance
+of the host.
+
+The example code how to use pinned memory in HIP showed at the following example.
+
+.. code-block:: cpp
+
+  #define ELEMENT_NUMBER 100
+
+  int *host_input, *host_output;
+  // Host allocation
+  hipHostMalloc((int **)&host_input, ELEMENT_NUMBER * sizeof(int));
+  hipHostMalloc((int **)&host_output, ELEMENT_NUMBER * sizeof(int));
+
+  // Host data preparation
+  for (int i = 0; i < ELEMENT_NUMBER; i++) {
+    host_input[i] = i;
+  }
+  memset(host_output, 0, ELEMENT_NUMBER * sizeof(int));
+
+  int *device_input, *device_output;
+
+  // Device allocation
+  hipMalloc((int **)&device_input, ELEMENT_NUMBER * sizeof(int));
+  hipMalloc((int **)&device_output, ELEMENT_NUMBER * sizeof(int));
+
+  // Device data preparation
+  hipMemcpy(device_input, host_input, ELEMENT_NUMBER * sizeof(int), cudaMemcpyHostToDevice);
+  hipMemset()
```

Comment:
Same as before: Please verify the whole code block

Reply:
Checked and fixed.